frontend/src/Utilities.js

```diff
@@ -24,6 +24,41 @@ export function getPrevSlot(day, slot) {
   };
 }
 
+function intersection() {
+  var result = [];
+  var lists;
+
+  if(arguments.length === 1) {
+    lists = arguments[0];
+  } else {
+    lists = arguments;
+  }
+
+  for(var i = 0; i < lists.length; i++) {
+    var currentList = lists[i];
+    for(var y = 0; y < currentList.length; y++) {
+        var currentValue = currentList[y];
+      if(result.indexOf(currentValue) === -1) {
+        var existsInAll = true;
+        for(var x = 0; x < lists.length; x++) {
+          if(lists[x].indexOf(currentValue) === -1) {
+            existsInAll = false;
+            break;
+          }
+        }
+        if(existsInAll) {
+          result.push(currentValue);
+        }
+      }
+    }
+  }
+  return result;
+}
+
+export function emptyAllDay(daySchedule) {
+  return(intersection(daySchedule))
```

Comment:
Unnecessary extra function. Implement the intersection function using `emptyAllDay` and explain what it does using comments

Reply:
✅ Implemented the intersection within `emptyAllDay`
✅ Added comments

Reply:
The comment can be better. Why do you need intersection? This will help anyone reading the code in future to understand why this thing is been done.

Reply:
Okay, I renamed the variables to be more descriptive and understandable and added a few lines of comments.